elide-core/src/main/java/com/yahoo/elide/core/EntityBinding.java

```diff
@@ -361,12 +363,37 @@ private void bindTriggerIfPresent(Class<? extends Annotation> annotationClass, A
             } catch (ReflectiveOperationException | IllegalArgumentException | SecurityException e) {
                 value = "";
             }
-            fieldsToTriggers.put(Pair.of(annotationClass, value), (Method) fieldOrMethod);
+
+            Method method = (Method) fieldOrMethod;
+            LifeCycleHook callback = (entity, scope, changes) -> {
+                try {
+                    if (changes.isPresent() && method.getParameterCount() == 2
+                            && method.getParameterTypes()[0].isInstance(scope)
+                            && method.getParameterTypes()[1].isInstance(changes.get())) {
+                        method.invoke(entity, scope, changes.get());
+                    } else if (method.getParameterCount() == 1 && method.getParameterTypes()[0].isInstance(scope)) {
+                        method.invoke(entity, scope);
+                    } else {
```

Comment:
/nit: maybe extract `method.getParameterCount()` and `method.getParameterTypes()` out into `bindTriggerIfPresent` and close over the extracted values so we're not redoing it every time we attempt to invoke this hook